packages/core/src/tools/create-file-stasher.js

```diff
@@ -237,6 +235,19 @@ const uploader = async (
   throw new Error(`Got ${response.status} - ${response.content}`);
 };
 
+const isUploadMaxSizeExceeded = (streamOrData, length) => {
```

Comment:
This function name is good if it just returns a boolean. But it can actually throw an error. So I'd suggest we rename it to something like `ensureUploadMaxSizeNotExceeded`.